### ISSUE TITLE
Delete genotyping protocol that has genotype data

### ISSUE THREAD

**Issue by nickpalladino** [solgenomics/sgn#4646]
There is a button on the UI (/breeders_toolbox/protocol/id) to delete genotyping protocols and all associated data but it doesn’t work when there is no collected genotyping data. In the backend (GenotypingProtocol.pm - genotyping_protocol_delete_GET) it’s running this query:
```
SELECT nd_experiment_id, genotype_id
FROM genotype
         JOIN nd_experiment_genotype USING(genotype_id)
         JOIN nd_experiment USING(nd_experiment_id)
         JOIN nd_experiment_protocol USING(nd_experiment_id)
WHERE nd_protocol_id = $protocol_id AND nd_experiment.type_id = $geno_cvterm_id;
```

The query isn’t returning any data which is leading to an error in the processing code following it. I believe it’s because there’s no actual genotyping data associated to the protocol in this case and the delete functionality is expecting there to be.

**Comment by lukasmueller** [comment]
May be the result of a failed loading process

**Comment by chris263** [comment]
Hey @isaak , did you have the opportunity to find out what might causing this issue?

**Comment by isaak** [comment]
I will check. The deletion does not work also on a protocol with data also.

**Comment by isaak** [comment]
The problems were delete privilege issues on genotype and nd_protocol tables in some dbs, and  with protocols that were partially uploaded: no genotypes loaded in the genotype table.. etc.
fixed #4737
